src/dso_hdr.cc

```diff
@@ -482,10 +482,15 @@ FileInfo DsoHdr::find_file_info(const Dso &dso) {
   //   or      /host/proc/<pid>/root/usr/local/bin/exe_file
   std::string proc_path = _path_to_proc + "/proc/" + std::to_string(dso._pid) +
       "/root" + dso._filename;
-  bool file_found = get_file_inode(proc_path.c_str(), &inode, &size);
-  if (file_found) {
+  if (get_file_inode(proc_path.c_str(), &inode, &size) && inode == dso._inode) {
     return FileInfo(proc_path, size, inode);
   }
+  // Try to find file in profiler mount namespace
+  if (get_file_inode(dso._filename.c_str(), &inode, &size) &&
```

Comment:
Now that we have the inode, should we not swap the order ?
Previous logic was that we were afraid of getting the wrong file. Now that you ensured that the files is good, we can simply rely on the dso filename as a priority ?

Reply:
Not sure about this:
* first lookup will fail only if process has already exited, which should not happen often
* in global mode, profiler mount namespace lookup will most probably fail all the time

IMHO the best solution would be:
* retrieve the target process mount namespace (upon exec?, lazily ?)
* if the target process mount namespace is the same as profiler's, do nothing special, open files in local namespace
* if target process mount namespace is different, open a file descriptor on `/proc/<pid>/root` if we don't have one yet (we need a mapping "mount namespace" -> fd), otherwise retrieve the existing one. Then use this file descriptor + openat when opening files fd for this process